agent/containers_internal_test.go

```diff
@@ -0,0 +1,206 @@
+package agent
+
+import (
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/testutil"
+	"github.com/coder/quartz"
+)
+
+// TestDockerCLIContainerLister tests the happy path of the
+// dockerCLIContainerLister.List method. It starts a container with a known
+// label, lists the containers, and verifies that the expected container is
+// returned. The container is deleted after the test is complete.
+func TestDockerCLIContainerLister(t *testing.T) {
+	t.Parallel()
+	if runtime.GOOS != "linux" {
+		t.Skip("creating containers on non-linux runners is slow and flaky")
+	}
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err, "Could not connect to docker")
+	testLabelValue := uuid.New().String()
+	ct, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "busybox",
+		Tag:        "latest",
+		Cmd:        []string{"sleep", "infnity"},
```

Comment:
Ha! TIL that `infblah` is a valid arg to `sleep` since it only scans for an `inf` suffix

Anyway I'm sure this was a typo, so for correctness' sake:

```suggestion
		Cmd:        []string{"sleep", "infinity"},
```